### ISSUE TITLE
stg crashes after pulling version 1.2

### ISSUE THREAD

**Issue by chucklever** [stacked-git/stgit#154]
This is on one of my Fedora 33 development systems. Another installed without incident.

```
$ git fetch origin
$ git reset --hard v1.2
$ sudo python3 setup.py install
$ stg version
Traceback (most recent call last):
  File "/usr/local/bin/stg", line 33, in <module>
    sys.exit(load_entry_point('stgit==1.2', 'console_scripts', 'stg')())
  File "/usr/local/bin/stg", line 25, in importlib_load_entry_point
    return next(matches).load()
StopIteration
$
```

I'm sure I'm just missing some Python dependency, but I have no pyFu .

**Comment by jpgrayson** [comment]
Hi @chucklever. Thank you for bravely trying StGit v1.2. I'm sorry that this out-of-the-box experience wasn't good.

One of the changes in v1.2 was that StGit now uses `setuptools` instead of `distutils` for packaging. Part of that is that at install-time, the `stg` executable is generated on-the-fly by `setuptools`. That traceback you see is from the generated `stg` installed to `/usr/local/bin` on your machine.

What this traceback tells me is that the `stgit` python package cannot be found by the `stg` entrypoint executable. It's not immediately clear why this would happen. My suspicion is that while the `stgit` package is installed to `/usr/local/lib/python3.X/site-packages`, but your system python executable does not look there for packages by default (i.e. because it is configured to only look in `/usr/lib/python3.X/...`).

A more sure-fire way of installing StGit from source is by using `pip`. E.g. from your stgit worktree:

```pip install --user .```

This will install to `~/.local`. You just need to make sure `~/.local/bin` is in your `PATH`. This is how I install development versions of StGit.

If you really need StGit to be installed for all users, then you might try `sudo pip install .`, but I'm not sure that will give a materially different outcome than `sudo make install`.

Yet another thing that may resolve your install to `/usr/local` would be to setup/update the `PYTHONPATH` environment variable to include `/usr/local/lib/python3.X/site-packages`. N.B. the "X" in "python3.X" needs to be the minor version of whatever python3 is installed on your system, e.g. "python3.9".

**Comment by chucklever** [comment]
Thanks for the quick response. I resolved the issue by wiping stgit out of /usr/local, and then using "sudo pip install .". Fortunately stgit is the only non-distribution-provided Python package I have on these systems.

```
$ stg version
Stacked Git 1.2
git version 2.33.0
Python version 3.9.6 (default, Jul 16 2021, 00:00:00) 
[GCC 10.3.1 20210422 (Red Hat 10.3.1-1)]
$
```

**Comment by chucklever** [comment]
Install worked, but "stg" still doesn't.

```
$ stg new blargh
Upgraded branch for-next to format version 5
Error: Unhandled exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/stgit/main.py", line 174, in _main
    ret = command.func(parser, options, args)
  File "/usr/local/lib/python3.9/site-packages/stgit/commands/new.py", line 99, in func
    cd = update_commit_data(
  File "/usr/local/lib/python3.9/site-packages/stgit/commands/common.py", line 513, in update_commit_data
    add_trailers(
  File "/usr/local/lib/python3.9/site-packages/stgit/utils.py", line 169, in add_trailers
    for trailer, user_value in trailers:
ValueError: too many values to unpack (expected 2)
$
```

**Comment by jpgrayson** [comment]
This bug is triggered by the use of `stgit.autosign`. Whoops. This is a gap in test coverage. I will repair and do a hot-fix release shortly.

In the meantime, you can work around the problem by using `stg new --sign ...` which will still cause the new patch to be signed, but without triggering this bug.

**Comment by zaneb** [comment]
> Fortunately stgit is the only non-distribution-provided Python package I have on these systems.

FWIW there is a distribution-provided version of stgit on Fedora (`sudo dnf install stgit`). It is a slightly older version though (F33 has 0.23 from last year).

**Comment by jpgrayson** [comment]
The issue regression with `stgit.autosign` is repaired in ae9ca309 and I released StGit v1.3 with this fix.

**Comment by chucklever** [comment]
> > Fortunately stgit is the only non-distribution-provided Python package I have on these systems.
> 
> FWIW there is a distribution-provided version of stgit on Fedora (`sudo dnf install stgit`). It is a slightly older version though (F33 has 0.23 from last year).

I like to use features in the latest git and stgit releases ;-)
